panther_analysis_tool/main.py

```diff
@@ -442,31 +491,63 @@ def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) ->
     client = get_client(args.aws_profile, "lambda")
-    analysis_id_list = args.analysis_id
-    payload: dict = {"deleteDetections": {"entries": []}}
 
-    # Validate the Detection that we are deleting exists in Panther
-    # If nothing was found bail out, otherwise prepare to delete what has been confirmed
+    datalake_function = "panther-snowflake-api"
+    if args.athena_datalake:
+        datalake_function = "panther-athena-api"
 
-    analysis_id_list = confirm_analysis_exists(args, analysis_id_list)
-    if len(analysis_id_list) == 0:
-        logging.error("No matching analysis found, exiting")
-        return 1, ""
+    # Delete function needs the query ID, required endpoint wont take a list
+    query_id_list = []
+    for query in query_list:
+        payload = {"listSavedQueries": {"pageSize": 1, "name": query}}
+        list_response = client.invoke(
+            FunctionName=datalake_function,
+            InvocationType="RequestResponse",
+            LogType="None",
+            Payload=json.dumps(payload),
+        )
+        api_response = json.loads(list_response["Payload"].read().decode("utf-8"))
+        api_saved_query_response = api_response.get("savedQueries")
+        if len(api_saved_query_response) == 0:
+            logging.warning("%s was not found, skipping...", query)
+            query_list.remove(query)
+            continue
+        query_id = api_saved_query_response[0]["id"]
+        query_id_list.append(query_id)
 
-    # Get user confirmation to delete
-    analysis_id_string = " ".join(analysis_id_list)
-    logging.warning("You are about to delete detections %s", analysis_id_string)
-    confirm = input("Continue? (y/n) ")
+    # Now we have query ids, lets delete them
 
-    if confirm.lower() != "y":
-        print("Cancelled")
+    if len(query_id_list) > 0:
+        payload = {
+            "deleteSavedQueries": {
+                "ids": query_id_list,
+                "userId": "00000000-0000-4000-8000-000000000000",
+                # The UserID is required by Panther for this API call, but we have no way of
+                # acquiring it, and it isn't used for anything. This is a valid UUID used by the
+                # Panther deployment tool to indicate this action was performed automatically.
+            }
+        }
+        delete_response = client.invoke(
+            FunctionName="panther-athena-api",
+            InvocationType="RequestResponse",
+            LogType="None",
+            Payload=json.dumps(payload),
+        )
+
+        if delete_response.get("ResponseMetadata").get("HTTPStatusCode") != 200:
+            error_payload = json.loads(list_response["Payload"].read().decode("utf-8"))
+            error_message = error_payload.get("errorMessage")
+            return 1, f"Error deleting queries, API error {error_message}"
+        logging.info("Queries %s have been deleted.", " ".join(query_list))
         return 0, ""
+    return 1, "No queries left to delete, exiting"
```

Comment:
should this return exit code of 0?

Reply:
I think 1 is appropriate because if you enter this function it's because either 1. Queries were passed on the CLI or 2. Related queries were found from detections passed CLI. If we get nothing to delete it feels like an error state to me